### ISSUE TITLE
 Enable PMD rule UseUnderscoresInNumericLiterals

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#6579]
https://pmd.github.io/latest/pmd_rules_java_codestyle.html#useunderscoresinnumericliterals

Part of #6536 

```
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes:1225
Rule:UseUnderscoresInNumericLiterals Priority:3 
Number 10000 should separate every third digit with an underscore.
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.api.DetailASTTest:288
 Rule:UseUnderscoresInNumericLiterals Priority:3 
Number 30000 should separate every third digit with an underscore.
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.api.JavadocTokenTypesTest:126
 Rule:UseUnderscoresInNumericLiterals Priority:3 
Number 10000 should separate every third digit with an underscore.
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.api.JavadocTokenTypesTest:127 
Rule:UseUnderscoresInNumericLiterals Priority:3 
Number 10071 should separate every third digit with an underscore.
...
```

**Comment by romani** [comment]
@pbludov , we need to review why our sevntu Check, does not catch it.
http://sevntu-checkstyle.github.io/sevntu.checkstyle/apidocs/com/github/sevntu/checkstyle/checks/coding/NumericLiteralNeedsUnderscoreCheck.html

**Comment by rnveach** [comment]
@romani Sevntu check is using default configuration and default length before a violation is 7 digits. The above violations only have 5 digits.
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/0a5c9480747cec7e3515462ca6b3503f87bc90b9/sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/NumericLiteralNeedsUnderscoreCheck.java#L176
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/0a5c9480747cec7e3515462ca6b3503f87bc90b9/sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/NumericLiteralNeedsUnderscoreCheckTest.java#L57-L61
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/7c747dc98f2479463ae6cf5e37d58a90eb3cb71b/sevntu-checks/src/test/resources/com/github/sevntu/checkstyle/checks/coding/InputNumericLiteralNeedsUnderscoreCheck.java#L27-L31

**Comment by pbludov** [comment]
We have following options:

- Enable the rule with `minLength` of 7 digits
- Fix all numbers and  set `NumericLiteralNeedsUnderscore` to 5 digits also
- Disable PMD rule in favor of our check

I vote for the last option.

**Comment by rnveach** [comment]
It is very rare we have such big numbers in our repo. Violations all refer to token types, which the numbers are generated and out of our control. If someone wants to add `_` so it is clear how many digits are in a number I am fine.

**Comment by romani** [comment]
I am voting for "Enable the rule with minLength of 7 digits" to match our Check and keep both of them active.
We disable pmd in favor of our Check when rule is misbehaving or conflicts a Check.

**Comment by rnveach** [comment]
Fix was merged
